src/pudl/analysis/demand_mapping.py

```diff
@@ -1,294 +1,786 @@
 
-    The function returns a stacked dataframe with mapping of each intermediate
-    geometry column with every non-intersecting source geometry column
-    quantifying the intersecting area.
+    Read the data into a goepandas.GeoDataFrame, and convert the data types as needed.
+    All columns are assigned nullable types.
 
     Args:
-        gdf_intermediate (GeoDataframe): intermediate geodataframe
-        gdf_source (GeoDataframe): usually source dataframe (i.e. demand)
-        gdf_intermediate_col (str): column to identify gdf_intermediate
-        gdf_source_col(str): column to identify for gdf_source
-        file_save (bool): save file option
+        pudl_settings (dict): A dictionary of PUDL settings, including the path to the
+            datastore.
 
     Returns:
-        pandas.DataFrame: DataFrame used to map and scale attribute from one
-        set of geometries to another.
-
-        Columns:
-            intermediate_index: unique index for intermediate geometry
-            source_index: unique index for source geometry
-            intermediate_intersection_fraction: area as fraction of
-            intermediate geometry
-            source_intersection_fraction: area as fraction of source geometry
+        geopandas.GeoDataFrame: containing the HIFLD planning area geometries and all
+        associated data columns.
 
     """
-    new_df = geopandas.overlay(
-        gdf_intermediate[[gdf_intermediate_col, geom_intermediate]],
-        gdf_source[[gdf_source_col, geom_source]],
-        how='intersection'
+    hifld_pa_url = "https://opendata.arcgis.com/datasets/7d35521e3b2c48ab8048330e14a4d2d1_0.gdb"
+    hifld_dir = pathlib.Path(pudl_settings["data_dir"]) / "local/hifld"
+    hifld_dir.mkdir(parents=True, exist_ok=True)
+    hifld_pa_zipfile = hifld_dir / "electric_planning_areas.gdb.zip"
+    hifld_pa_gdb_dir = hifld_dir / "electric_planning_areas.gdb"
+
+    if not hifld_pa_gdb_dir.is_dir():
+        logger.warning("No Planning Area GeoDB found. Downloading from HIFLD.")
+        # Download to appropriate location
+        pudl.helpers.download_zip_url(hifld_pa_url, hifld_pa_zipfile)
+        # Unzip because we can't use zipfile paths with geopandas
+        with zipfile.ZipFile(hifld_pa_zipfile, 'r') as zip_ref:
+            zip_ref.extractall(hifld_dir)
+            # Grab the UUID based directory name so we can change it:
+            extract_root = hifld_dir / \
+                pathlib.Path(zip_ref.filelist[0].filename).parent
+        extract_root.rename(hifld_pa_gdb_dir)
+    else:
+        logger.info("We've already got the planning area GeoDB.")
+
+    gdf = (
+        geopandas.read_file(hifld_pa_gdb_dir)
+        .assign(
+            SOURCEDATE=lambda x: pd.to_datetime(x.SOURCEDATE),
+            VAL_DATE=lambda x: pd.to_datetime(x.VAL_DATE),
+            ID=lambda x: pd.to_numeric(x.ID),
+            NAICS_CODE=lambda x: pd.to_numeric(x.NAICS_CODE),
+            YEAR=lambda x: pd.to_numeric(x.YEAR),
```

Comment:
Yeah, I didn't try and change the column names in HIFLD since we're not really integrating it as a dataset. It's just for comparison.

The `pd.to_datetime()` and `pd.to_numeric()` functions do more complex interpretation of values than simply assigning a type. The columns getting `assign()`-ed here were unable to be cast to their respective data types without this pre-processing step.